server-rlimit-nix.go

```diff
@@ -42,10 +42,7 @@ func setMaxOpenFiles() error {
 		return err
 	}
 	err = syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
-	if err != nil {
-		return err
-	}
-	return nil
+	return err
```

Comment:
Further simplification is 

`` return syscall.Getrlimit(syscal.RLIMIT_NOFILE, &rLimit) ``